vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/supported_subdivisions.cc

```diff
@@ -0,0 +1,81 @@
+      supported_subdivision_infos(
+          {{"US",
+            {
+                // United States of America
+                {"US-AL", "Alabama"},        {"US-AK", "Alaska"},
+                {"US-AZ", "Arizona"},        {"US-AR", "Arkansas"},
+                {"US-CA", "California"},     {"US-CO", "Colorado"},
+                {"US-CT", "Connecticut"},    {"US-DE", "Delaware"},
+                {"US-FL", "Florida"},        {"US-GA", "Georgia"},
+                {"US-HI", "Hawaii"},         {"US-ID", "Idaho"},
+                {"US-IL", "Illinois"},       {"US-IN", "Indiana"},
+                {"US-IA", "Iowa"},           {"US-KS", "Kansas"},
+                {"US-KY", "Kentucky"},       {"US-LA", "Louisiana"},
+                {"US-ME", "Maine"},          {"US-MD", "Maryland"},
+                {"US-MA", "Massachusetts"},  {"US-MI", "Michigan"},
+                {"US-MN", "Minnesota"},      {"US-MS", "Mississippi"},
+                {"US-MO", "Missouri"},       {"US-MT", "Montana"},
+                {"US-NE", "Nebraska"},       {"US-NV", "Nevada"},
+                {"US-NH", "New Hampshire"},  {"US-NJ", "New Jersey"},
+                {"US-NM", "New Mexico"},     {"US-NY", "New York"},
+                {"US-NC", "North Carolina"}, {"US-ND", "North Dakota"},
+                {"US-OH", "Ohio"},           {"US-OK", "Oklahoma"},
+                {"US-OR", "Oregon"},         {"US-PA", "Pennsylvania"},
+                {"US-RI", "Rhode Island"},   {"US-SC", "South Carolina"},
+                {"US-SD", "South Dakota"},   {"US-TN", "Tennessee"},
+                {"US-TX", "Texas"},          {"US-UT", "Utah"},
+                {"US-VT", "Vermont"},        {"US-VA", "Virginia"},
+                {"US-WA", "Washington"},     {"US-WV", "West Virginia"},
+                {"US-WI", "Wisconsin"},      {"US-WY", "Wyoming"},
+            }},
+           {"CA",
+            {
+                // Canada
+                {"CA-AB", "Alberta"},
+                {"CA-BC", "British Columbia"},
+                {"CA-MB", "Manitoba"},
+                {"CA-NB", "New Brunswick"},
+                {"CA-NS", "Nova Scotia"},
+                {"CA-ON", "Ontario"},
+                {"CA-QC", "Quebec"},
+                {"CA-SK", "Saskatchewan"},
+            }}});
+
+  return *supported_subdivision_infos;
+}
+
+bool IsSupportedLocale(const std::string& locale) {
+  const std::string country_code = brave_l10n::GetCountryCode(locale);
+
+  const auto iter = GetSupportedSubdivisionInfos().find(country_code);
+  return iter != GetSupportedSubdivisionInfos().cend();
+}
+
+bool IsSupportedSubdivisionCode(const std::string& country_code,
+                                const std::string& subdivision_code) {
+  const auto iter = GetSupportedSubdivisionInfos().find(country_code);
+  if (iter == GetSupportedSubdivisionInfos().cend()) {
+    return false;
+  }
+
+  const auto& subdivisions = iter->second;
+  return subdivisions.find(subdivision_code) != subdivisions.cend();
+}
```

Comment:
`IsSupported...` could be moved to a subdivision_code_util and then his file could be supported_subdivision_codes